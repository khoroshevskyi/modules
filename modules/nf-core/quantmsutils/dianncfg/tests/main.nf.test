nextflow_process {

    name "Test Process QUANTMSUTILS_DIANNCFG"
    script "../main.nf"
    process "QUANTMSUTILS_DIANNCFG"
    config "./nextflow.config"

    tag "modules"
    tag "modules_nfcore"
    tag "quantmsutils"
    tag "quantmsutils/dianncfg"

    test("proteomics - generate DIA-NN config") {

        when {
            process {
                """
                input[0] = Channel.of([
                    [id:'test_exp'],
                    'Trypsin',
                    'Carbamidomethyl (C)',
                    'Oxidation (M)'
                ])
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }

    test("proteomics - generate DIA-NN config - stub") {

        options "-stub"

        when {
            process {
                """
                input[0] = Channel.of([
                    [id:'test_exp'],
                    'Trypsin',
                    'Carbamidomethyl (C)',
                    'Oxidation (M)'
                ])
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }
}
