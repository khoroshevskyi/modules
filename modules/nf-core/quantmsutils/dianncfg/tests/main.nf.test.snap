{
    "proteomics - generate DIA-NN config": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test_exp"
                        },
                        "diann_config.cfg:md5,0d6e936c98e6e3cfad326be8c54a80c4"
                    ]
                ],
                "1": [
                    "GENERATE_DIANN_CFG.log:md5,23eb245baac51a239ce6cf022643a05b"
                ],
                "2": [
                    "versions.yml:md5,b4ffbe493e0f58a476654ec598ef4385"
                ],
                "diann_cfg": [
                    [
                        {
                            "id": "test_exp"
                        },
                        "diann_config.cfg:md5,0d6e936c98e6e3cfad326be8c54a80c4"
                    ]
                ],
                "log": [
                    "GENERATE_DIANN_CFG.log:md5,23eb245baac51a239ce6cf022643a05b"
                ],
                "versions": [
                    "versions.yml:md5,b4ffbe493e0f58a476654ec598ef4385"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.6"
        },
        "timestamp": "2025-10-06T12:20:39.51872"
    },
    "proteomics - generate DIA-NN config - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test_exp"
                        },
                        "diann_config.cfg:md5,bb094d79a4014faab1368aab577a4ea0"
                    ]
                ],
                "1": [
                    "GENERATE_DIANN_CFG.log:md5,d41d8cd98f00b204e9800998ecf8427e"
                ],
                "2": [
                    "versions.yml:md5,b4ffbe493e0f58a476654ec598ef4385"
                ],
                "diann_cfg": [
                    [
                        {
                            "id": "test_exp"
                        },
                        "diann_config.cfg:md5,bb094d79a4014faab1368aab577a4ea0"
                    ]
                ],
                "log": [
                    "GENERATE_DIANN_CFG.log:md5,d41d8cd98f00b204e9800998ecf8427e"
                ],
                "versions": [
                    "versions.yml:md5,b4ffbe493e0f58a476654ec598ef4385"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.6"
        },
        "timestamp": "2025-10-06T12:20:45.611066"
    }
}